server/lib/models/email-verifications.js

```diff
@@ -0,0 +1,44 @@
+import db from '../db'
+
+export async function addEmailVerificationCode(userId, email, code, ip) {
+  const { client, done } = await db()
+  const query = `
+      INSERT INTO email_verifications
+      (user_id, email, verification_code, request_time, request_ip)
+      VALUES ($1, $2, $3, $4, $5)
+  `
+  const params = [userId, email, code, new Date(), ip]
+
+  try {
+    await client.query(query, params)
+  } finally {
+    done()
+  }
+}
+
+export async function useEmailVerificationCode(id, email, code) {
+  const query = `
+    UPDATE users u
```

Comment:
This query should ensure that the user's email matches the one we're verifying for